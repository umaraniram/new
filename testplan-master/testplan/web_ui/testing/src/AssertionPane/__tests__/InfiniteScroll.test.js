import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import InfiniteScroll from '../InfiniteScroll';

function defaultProps() {
  return {
    globalIsOpen: false,
    id: "22758cc5-8a89-472b-bf67-b64dbc2c0b40",
    items: [],
    resetGlobalIsOpen: jest.fn()
  };
}


describe('InfiniteScroll', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<InfiniteScroll {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});