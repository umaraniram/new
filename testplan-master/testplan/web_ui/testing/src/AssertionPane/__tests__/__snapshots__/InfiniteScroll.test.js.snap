// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll shallow renders the correct HTML structure 1`] = `
<Scrollbars
  autoHeight={false}
  autoHeightMax={200}
  autoHeightMin={0}
  autoHide={true}
  autoHideDuration={200}
  autoHideTimeout={1000}
  hideTracksWhenNotNeeded={false}
  onScroll={[Function]}
  renderThumbHorizontal={[Function]}
  renderThumbVertical={[Function]}
  renderTrackHorizontal={[Function]}
  renderTrackVertical={[Function]}
  renderView={[Function]}
  tagName="div"
  thumbMinSize={30}
  universal={false}
>
  <div
    style={
      Object {
        "paddingRight": "2rem",
      }
    }
  />
</Scrollbars>
`;
